ДЗ спринт №11

morechiba opened the pull request.

morechiba committed:
подключение gson и retrofit

morechiba committed:
Запрос к ITunes. Начало. Ошибочные запросы - начало верстки.

morechiba committed:
Верстка placeholder для ошибочных запросов.

morechiba committed:
Картинки в placeholder с ресурсным цветом (для темной темы).

morechiba committed:
Поиск и placeholder "не найдено" и "нет связи" в работе.

morechiba committed:
Возврат кнопки стирания в EditText. При нажатии на эту кнопку добавлено стирание списка треков.

morechiba committed:
Поиск вынесен в отдельную функцию. По клику на кнопку "Обновить" при ошибке поиска - поиск повторяется через функцию.

ArturNurtdinov commented on app/build.gradle.kts:
```diff
@@ -42,9 +42,13 @@ dependencies {
     implementation(libs.material)
     implementation(libs.androidx.activity)
     implementation(libs.androidx.constraintlayout)
+    implementation("com.google.code.gson:gson:2.10")
+    implementation("com.squareup.retrofit2:retrofit:2.11.0")
+    implementation("com.squareup.retrofit2:converter-gson:2.11.0")
```
Все новые зависимости стоит делать через version catalog механизм - файл `libs.versions.toml`, чтобы все внешние зависимости проекта находились в одном месте - так удобнее будет с ними работать, а также все модули проекта будут зависеть от одних и тех же версий сторонних зависимостей

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/SearchActivity.kt:
```diff
@@ -30,6 +38,15 @@ class SearchActivity : AppCompatActivity() {
+        .build()
+
+    private val itunesSearch = retrofit.create(ITunesApi::class.java)
+    val trackList = ArrayList<Track>()
```
Рекомендуется использовать котлиновские интерфейсы `MutableList` и `List` для работы со списками. Создать такие списки можно с помощью функций `mutableListOf()` и `listOf()` соответственно

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/SearchActivity.kt:
```diff
@@ -85,49 +106,73 @@ class SearchActivity : AppCompatActivity() {
+                        response: Response<SearchResponse>
+                    ) {
+
+                        if (response.code() == 200) {
```
Лучше воспользоваться свойством `isSuccessful` для проверки успешности выполнения запроса, чтобы не хардкодить коды ответов http запросов. Плюс `isSuccessful` проверяет и другие коды, обозначающие, что запрос был успешно выполнен - 200-299

ArturNurtdinov commented on app/src/main/java/com/tagomago/playlistmaker/SearchActivity.kt:
```diff
@@ -85,49 +106,73 @@ class SearchActivity : AppCompatActivity() {
+
+                        if (response.code() == 200) {
+                            if (response.body()?.results?.isNotEmpty() == true) {
+                                trackList.addAll(response.body()?.results!!)
```
`response.body()?.results` лучше вынести в отдельную переменную, чтобы избавиться от лишнего использования операторов `?` и `!!`

ArturNurtdinov reviewed (commented):
Очень хорошая работа, молодец!

morechiba committed:
1. Проверка на 200 ответ заменена на успешный ответ (коды 200-299) - isSuccessful.
2. response.body()?.results вынесен в отдельную переменную, чтобы избавиться от лишнего использования операторов ? и !!

morechiba committed:
ArrayList заменен на MutableList

morechiba committed:
новые файлы откуда-то
